core/templates/dev/head/domain/collection/CollectionRightsBackendApiService.js

```diff
@@ -32,12 +54,33 @@ oppia.factory('CollectionRightsBackendApiService', [
           version: collectionVersion,
           is_public: isPublic
         };
-        $http.put(collectionRightsUrl, putParams).then(function() {
-          // TODO(bhenning): Consolidate the backend rights domain objects and
-          // implement a frontend activity rights domain object. The rights
-          // being passed in here should be used to create one of those objects.
-          if (successCallback) {
-            successCallback();
+        $http.put(collectionRightsUrl, putParams).then(function(response) {
+          // Check if the response from the backend does not contradict
+          // putParams.
+          var isPrivateResponse = (response.data.status === 'private');
+          if (isPrivateResponse === isPublic) {
+            if (isPublic) {
+              $log.error(
+                'Backend indicated a collection was successfully ' +
+                'published, but response.data.is_private returned true.');
+            } else {
+              $log.error(
+                'Backend indicated a collection was successfully ' +
+                'unpublished, but response.data.is_private returned false.');
+            }
+            if (errorCallback) {
+              errorCallback(response.data);
+            }
+          } else {
+            // An additional fetch is required because the GET response object
```

Comment:
Here is my original comment, and the file which I made the comment on: https://github.com/oppia/oppia/pull/2867/files/456d2a5ec7652b1d5700168d7392c61d42c96a3b#diff-974a2e9766128d912e24f870f8fad7d0R212

I'm fine with modifying the dict returned by the backend, as @BenHenning is suggesting. However, I don't think we should take the "rights" object in the backend and change its properties, since then we start introducing multiple meanings for the same concept.